core/lib.py

```diff
@@ -0,0 +1,58 @@
+import contextlib
+import io
+import os
+from typing import Optional, cast
+
+from pathlib import Path
+
+import settings
+
+get_active_session: Optional[callable] = None

```

Comment:
This is a hack so we can get the session name without needing to pass around pacu_main everywhere.